middlewares/prometheus_test.go

```diff
@@ -44,4 +46,24 @@ func TestPrometheus(t *testing.T) {
 	if !strings.Contains(body, latencyName) {
 		t.Errorf("body does not contain request duration entry '%s'", reqsName)
 	}
+
+	// Register the same metrics again
+	metricsMiddlewareBackend = NewMetricsWrapper(NewPrometheus("test", &types.Prometheus{}))
+	n = negroni.New()
+	n.Use(metricsMiddlewareBackend)
+	n.UseHandler(r)
+
+	recorder = httptest.NewRecorder()
+	n.ServeHTTP(recorder, req2)
+
+	body = recorder.Body.String()
+	re := regexp.MustCompile(`(?m)traefik_requests_total.*?(\d)$`)
+	if re.FindStringSubmatch(body)[1] != "2" {
+		t.Error("body does not contain correct value for total requests")
+	}
+
+	metrics, _ := prometheus.DefaultGatherer.Gather()
+	if metrics[len(metrics)-1].Metric[0].GetCounter().GetValue() != 3 {
```

Comment:
At the time when request 3 happens the counter still has value 2 and this value is outputted to body (which is used in comparison on line 61). Only after that the counter increases. (But this doesn't really matter because I will remove the regex check.)